test/tools/yulopti.cpp

```diff
@@ -87,37 +88,47 @@ class YulOpti
 		}.printErrorInformation(_errors);
 	}
 
-	void parse(string const& _input)
+	void parse(string const& _input, string const& _objectPath)
 	{
 		ErrorList errors;
 		ErrorReporter errorReporter(errors);
-		CharStream _charStream(_input, "");
+		CharStream charStream(_input, "");
+
 		try
 		{
-			m_ast = yul::Parser(errorReporter, m_dialect).parse(_charStream);
-			if (!m_ast || !errorReporter.errors().empty())
+			ObjectParser parser(errorReporter, m_dialect);
+
+			auto scanner = make_shared<Scanner>(charStream);
+
+			if (!m_inputIsCodeBlock && scanner->currentToken() == Token::LBrace)
 			{
-				cerr << "Error parsing source." << endl;
-				printErrors(_charStream, errors);
-				throw std::runtime_error("Could not parse source.");
+				m_inputIsCodeBlock = true;
+
+				if (!_objectPath.empty())
+					throw runtime_error("Object path argument cannot be used. Input is a code block.");
 			}
-			m_analysisInfo = make_unique<yul::AsmAnalysisInfo>();
-			AsmAnalyzer analyzer(
-				*m_analysisInfo,
-				errorReporter,
-				m_dialect
-			);
-			if (!analyzer.analyze(*m_ast) || !errorReporter.errors().empty())
+
+			shared_ptr<Object> object = parser.parse(scanner, false);
+
+			if (!object || !errorReporter.errors().empty())
 			{
-				cerr << "Error analyzing source." << endl;
-				printErrors(_charStream, errors);
-				throw std::runtime_error("Could not analyze source.");
+				cerr << "Error parsing source." << endl;
+				printErrors(charStream, errors);
+				throw runtime_error("Could not parse source.");
+			}
+
+			m_object = Object::objectAt(object, _objectPath);
+
+			if (m_object == nullptr) {
+				throw runtime_error("Assembly object not found.");
 			}
```

Comment:
```suggestion
			if (m_object == nullptr)
				throw runtime_error("Assembly object not found.");
```
Also, maybe we should throw something a different type such as `util::Exception` or `yul::YulException`  instead of `runtime_error`.